### PR TITLE
va-select: add width prop to set width of select element

### PR DESCRIPTION
**Pull request by it-harrison**
## Chromatic

https://2644-va-select-width--65a6e2ed2314f7b8f98609d8.chromatic.com

---

## Description
This PR adds a `width` prop to `va-select` which allows the user to set the width of the underlying `select` element in the same way that the width is set on `va-text-input`.
Closes [2644](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2644)

## QA Checklist
- [ ] Component maintains 1:1 parity with design mocks
- [ ] Text is consistent with what's been provided in the mocks
- [ ] Component behaves as expected across breakpoints
- [ ] Accessibility expert has signed off on code changes (if applicable. If not applicable provide reason why)
- [ ] Designer has signed off on changes (if applicable. If not applicable provide reason why)
- [ ] Tab order and focus state work as expected
- [ ] Changes have been tested against screen readers (if applicable. If not applicable provide reason why)
- [ ] New components are covered by e2e tests; updates to existing components are covered by existing test suite
- [ ] Changes have been tested in vets-website using Verdaccio (if applicable. If not applicable provide reason why)

## Screenshots
Screenshot 2024-08-29 at 11 49 31 AM

## Acceptance criteria
- [ ] QA checklist has been completed
- [ ] Screenshots have been attached that cover desktop and mobile screens

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Review by Andrew565 — APPROVED** [review]

**Comment by jamigibbs** [comment]
@danbrady @babsdenney @LWWright7 Do y'all need or want this update added to Figma?

**Comment by danbrady** [comment]
@jamigibbs Yes! Thanks for bringing that to our attention. @caw310 Can you create two tickets please? One design ticket for adding widths to Select in Figma and one documentation ticket to add these examples (and props) to the Select page on the doc site? Thanks!

This looks good to me. I agree with Jami that the width examples should show the same why the text input does.

**Comment by caw310** [comment]
@danbrady 
Here are the tickets to update Figma and guidance. 

[Update select component in Figma - Design](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3250)
[Update guidance for select component - Documentation](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3251)

Tentatively in sprints 10 and 11.

File: packages/web-components/src/components.d.ts [hunk 1]
```diff
@@ -1371,6 +1371,10 @@ export namespace Components {
           * Selected value (will get updated on select).
          */
         "value"?: string;
+        /**
+          * Displays the select at a specific width. Accepts 2xs (4ex), xs (7ex), sm or small (10ex), md or medium (20ex), lg (30ex), xl (40ex), and 2xl (50ex).
```
[the remark this answers is not in the document]
Comment (it-harrison) [reply packages/web-components/src/components.d.ts:1375]: @jamigibbs - story updated:
Screenshot 2024-08-29 at 8 49 34 PM

[the remark this answers is not in the document]
Comment (jamigibbs) [reply packages/web-components/src/components.d.ts:1375]: hmmm yeah those first two look a little strange right? 🤔 

Screenshot 2024-08-30 at 9 57 16 AM

@danbrady From a design perspective, do you think `2xs` and `xs` are even viable sizes for this component style with the select value? If not, maybe we should not allow those sizes to be used.

**Pull request merged**